crates/lib/src/qpu/api.rs

```diff
@@ -202,6 +205,62 @@ pub struct ExecutionOptions {
     #[doc = "The timeout to use for the request, defaults to 30 seconds. If set to `None`, then there is no timeout."]
     #[builder(default = "Some(Duration::from_secs(30))")]
     timeout: Option<Duration>,
+    #[doc = "Options avaialable when executing a job on a QPU, particular to the execution service's API."]
+    #[builder(default = "None")]
+    api_options: Option<ApiExecutionOptions>,
+}
+
+/// Options avaialable when executing a job on a QPU, particular to the execution service's API.
+///
+/// Use [`Default`] to get a reasonable set of defaults, or start with [`ApiExecutionOptionsBuilder`]
+/// to build a custom set of options.
+#[derive(Builder, Clone, Debug, Default)]
+#[allow(clippy::module_name_repetitions)]
+pub struct ApiExecutionOptions {
+    /// the inner proto representation
+    inner: InnerApiExecutionOptions,
+}
+
+impl PartialEq for ApiExecutionOptions {
+    /// implement Eq
+    fn eq(&self, other: &Self) -> bool {
+        let InnerApiExecutionOptions {
+            bypass_settings_protection,
+        } = self.inner;
+        bypass_settings_protection == other.inner.bypass_settings_protection
+    }
+}
```

Comment:
`InnerApiExecutionOptions` implements `PartialEq` but not `Eq`, so you should be able to derive `PartialEq` instead of this manual implementation.

Should we consider removing the `Eq` (derived) impl on the types containing this one instead? `==` will work as long as `PartialEq` is implemented.

Reply:
Fixed this with https://github.com/rigetti/qcs-sdk-rust/pull/415/commits/7a3f4fd0723218329d0dd1e1fc709b6ef739a3cb - left in the following and things are working

```rs
impl Eq for ApiExecutionOptions {}
```